ch08.adoc

```diff
@@ -675,3 +675,130 @@ The data creator shall specify the floating-point arithmetic precision used duri
+[caption="Example 8.9. "]
+.Quantization performed by Granular BitGroom algorithm in NCO
+====
+Quantization of different variables to different levels often makes good scientific sense. Here the pressure variable **`ps`** has four significant digits of precision while the temperature variable **`ts`** retains only three significant digits. 
+----
+  variables:
+    char quantization_info ;
+      quantization_info:algorithm = "granular_bitround" ;
+      quantization_info:implementation = "NCO version 5.2.7" ;
+
+    float ps(time,lat,lon) ;
+      ps:standard_name = "surface_air_pressure" ;
+      ps:units = "Pa" ;
+      ps:quantization = "quantization_info" ;
+      ps:quantization_nsd = 4 ;
+
+    float ts(time) ;
+      ts:standard_name = "surface_temperature" ;
+      ts:units = "K" ;
+      ts:quantization = "quantization_info" ;
+      ts:quantization_nsd = 3 ;
+----
+Both variables were quantized by the same algorithm and so utilize the same quantization variable. **`quantization_info`** reveals that the Granular BitRound algorithm in NCO performed the quantization. Since the netCDF library did not perform the quantization, there is no system-defined long quantization attribute.
+====
+
+[[quantization-algorithms-description, Section 8.4.3, "Description of Quantization Algorithms"]]
+==== Description of quantization algorithms
+
+This section briefly describes and contrasts each recognized **`quantize`** algorithm and points to further documentation.
+BitRound is also called the "round-to-nearest" method [<<KRD21>>] and the "half-to-even" method [<<Kou21>>].
+This is the default [<<IEEE_754>>] rounding method and is bias-free and conservative for random distributions of numbers.
+BitRound is preferred when the number of significant bits (NSB) to retain is known.
+
+The other **`quantize`** algorithms guarantee to preserve a given number of significant (base-10 representation) digits (NSD).
+Their quantization errors never exceed half of the unit value at the NSD decimal place [<<Zen16>>].
+BitGroom [<<Zen16>>] appeared first, though is now known to be suboptimal in accuracy [<<Kou21>>] and in compressibility compared to later methods.
+DigitRound [<<DCG19>>] has superior compressibility for a given NSD compared to BitGroom.
+Granular BitGroom combines the DigitRound approach for compressibility with the BitRound approach for quantization.
+Granular BitGroom and DigitRound are both good choices when the NSD to retain is known.
+
+The netCDF C and Fortran libraries can directly invoke BitRound, BitGroom, and Granular BitRound [<<NetCDF>>].
+The netCDF library attaches a long, system-defined attribute to every data variable that it quantizes, such as 
+**`_QuantizeBitRoundNumberOfSignificantBits = 9`** in Example 8.8.
+The leading underscore indicates that the netCDF library wrote this attribute [<<NUG>>].
+Any data variable that has the library-defined attribute should, in addition, contain the corresponding CF metadata.
```

Comment:
I think we must "must" this, to avoid a possible ambiguity.
```suggestion
Any variable that has the library-defined attribute must, in addition, contain the corresponding CF metadata.
```